isodesign/process/calculation.py

```diff
@@ -1,106 +1,87 @@
 """ Module for calculation """
 import os
-from itertools import product
-import pandas as pd
+from itertools import product, combinations
+import math
 
+import pandas as pd
+import numpy as np
 
 class Tracer:
     """ 
-    This class is for the initiation of tracers.
+    Class for the initiation of tracers.
+
     """
 
-    def __init__(self, name, labeling):
+    def __init__(self, name, labeling, step, lower_bound, upper_bound):
         self.name = name
         self.labeling = labeling
         self.num_carbon = len(self.labeling)
-
+        self.step = step
+        self.lower_bound = lower_bound
+        self.upper_bound = upper_bound
+        self.fraction = [fraction for fraction in range(self.lower_bound, self.upper_bound + self.step, self.step)]
+       
     def __len__(self):
-        """ 
-        Count the carbon number by using the labeling.
-        """
         return self.num_carbon
 
     def __repr__(self) -> str:
         return f"Molecule name: {self.name}, Number of associated carbon(s) : {self.num_carbon}"
-
-
-def _generate_frac_comb(molecules, step=10):
-    """
-    Generate combinations of different molecular tracers proportions
-
-    :param molecules: list of tracer molecules
-    :param step: step for proportions to test
     
-    :return: list of tuples containing proportions of each tracer in mix (sum must be equant to end param value)
-    """
-
-    all_fracs = []
-    for _ in range(len(molecules)):
-        fractions = [data for data in range(0, 100 + step, step)]
-        all_fracs.append(fractions)
-
-    # list containing all the combination inside a molecule family
-    mixes = [frac_list for frac_list in product(*all_fracs) if sum(frac_list) == 100]
-    return mixes
-
-
-def generate_mixes_comb(*args):
-    """ 
-    Generate cominations between two molecular mixes of tracers
+    @property
+    def lower_bound(self):
+        return self._lower_bound
     
-    :param args:list of mixes to combine together
-
-    :return: list of all combinations of the tracer mixes
-    """
-    if len(args) == 1:
-        return _generate_frac_comb(args[0])
-    return list(product(*[_generate_frac_comb(mix) for mix in args]))
-
-def generate_file(carbon_source, *args):
-    """
-    Generating .linp files in function of all combination for one or two mixe(s)
-
-    :param carbon_source: list of tracer molecules
-    :param carbon_source_2: list of tracer molecules
+    @lower_bound.setter
+    def lower_bound(self, value):
+        if value < 0:
+            raise ValueError(f"{value} isn't correct. Use a positive value.")
```

Comment:
Je mettrais plutôt "Lower bound for a tracer proportion must be a positive number." car c'est plus explicite